### PR TITLE
feat(localstatequery): add GetAccountState query

### PR DESCRIPTION
**Pull request by chrisguiney**
## Summary

`GetAccountState` (Shelley/Conway block sub-query **29**) was a gap in the local-state-query registry — indices 22, 29 and 33 were unregistered. Because the server-side decoder couldn't map sub-query 29, any client that issued it had its `LedgerStateQuery` connection torn down. `cardano-cli` issues `GetAccountState` while balancing a transaction (`transaction build`), so this gap broke that flow for any node built on gOuroboros (e.g. Dingo).

## Changes

- Add `QueryTypeShelleyAccountState = 29` and the `ShelleyAccountStateQuery` type (a no-arg `simpleQueryBase`).
- Register it in the block sub-query decode map so the server can decode it.
- Add the result types `AccountState` (`{treasury, reserves}`) and `AccountStateResult`, plus a `Client.GetAccountState()` method.

## Wire format

The result is the account-state pots wrapped in the single-element result array — on the wire `[ [treasury, reserves] ]`. Both values are **signed**, since `Coin` is an `Integer` in the ledger; a misconfigured network can drive reserves negative (observed on the devnet's cardano-node). This was confirmed by capturing cardano-node's actual `GetAccountState` reply on a Conway devnet and matching it byte-for-byte.

## Tests

`TestShelleyAccountStateQueryBuildShelleyQuery` (query envelope encoding) and `TestAccountStateResultRoundTrip` (result round-trip incl. negative reserves). `go test ./protocol/localstatequery/...`, `gofmt`, `go vet` all pass.

This unblocks `cardano-cli transaction build` against gOuroboros-based nodes — verified end-to-end with Dingo: the build now completes and produces a valid `Tx ConwayEra` (see the companion Dingo change).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Add `GetAccountState` (Shelley/Conway sub-query 29) to the local-state-query registry and client, fixing connection drops when `cardano-cli` requests account state during `transaction build`. The client now gates this query to Conway era and returns a clear error on pre-Conway chains.

- **New Features**
  - Register `QueryTypeShelleyAccountState = 29` and `ShelleyAccountStateQuery` in the decode map.
  - Add result types: `AccountState` (signed `treasury`, `reserves`) and `AccountStateResult`.
  - Add `Client.GetAccountState()` with a Conway-era guard.
  - Wire format: result is `[ [treasury, reserves] ]`.

<sup>Written for commit 4e9a47121bfec5b1c36d1b463818377a865e0ad5. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/blinklabs-io/gouroboros/pull/1818?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

## Release Notes

* **New Features**
  * Added new account state query method to the LocalStateQuery client API for Shelley-era queries.
  * Enables retrieval of current treasury and reserves information for governance operations.

* **Tests**
  * Added test coverage validating CBOR encoding/decoding of account state queries and results.
  * Includes round-trip serialization testing with edge cases.

### CODE REVIEW

File: protocol/localstatequery/client.go [hunk 1]
```diff
@@ -729,6 +729,31 @@ func (c *Client) GetStakePools() (*StakePoolsResult, error) {
...
+		currentEra,
+		QueryTypeShelleyAccountState,
+	)
+	var result AccountStateResult
+	if err := c.runQuery(query, &result); err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
```
[the remark this answers is not in the document]
Comment (chrisguiney) [reply protocol/localstatequery/client.go:762]: Good catch — added the Conway-era guard in 4e9a471, matching `GetConstitution`/`GetGovState`/`GetDRepState`.

File: protocol/localstatequery/client.go [hunk 2]
```diff
@@ -729,6 +729,31 @@ func (c *Client) GetStakePools() (*StakePoolsResult, error) {
...
+			"role", "client",
+			"connection_id", c.callbackContext.ConnectionId.String(),
+		)
+	c.busyMutex.Lock()
+	defer c.busyMutex.Unlock()
+	currentEra, err := c.getCurrentEra()
+	if err != nil {
+		return nil, err
+	}
+	query := buildShelleyQuery(
```
[the remark this answers is not in the document]
Comment (chrisguiney) [reply protocol/localstatequery/client.go:752]: Good catch — added the Conway-era guard in 4e9a471, matching `GetConstitution`/`GetGovState`/`GetDRepState`.

**Review by wolf31o2 — APPROVED** [review]
